api/tests/Feature/PoolApplicationTest.php

```diff
@@ -163,6 +165,7 @@ public function testApplicationCreation(): void
             'closing_date' => config('constants.far_future_date'),
             'advertisement_language' => ApiEnums::POOL_ADVERTISEMENT_ENGLISH, // avoid language requirements
         ]);
+        ScreeningQuestion::truncate();
```

Comment:
It was failing without it, but maybe it was something to do with how I was using the factories initially that was fixed at some point and these were silently doing nothing 🤷‍♀️ It is working after [remove screening question truncation](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6382/commits/1622b5b79b4aa001ee6a0944682c35d7aaf1b23a) 👍 

I guess how we do our factories is fairly subjective but I am of the opion that we should not be automatically be creating relationships in factories unless they are required in some way (a `PoolCandidate` must always been attached to a `Pool`). These relationships are entirely optional and we would need to delete them after creation if we want to seed a either a `Pool` or `PoolCandidate` without them and their responses, which to me seems odd 😅 .

I wonder if we could randomly seed 0-3 instead of always having 3 with all pool candidates responding to all. Though, I would like to see us be more explicit about the data seed rather than have factories always seed the most optimistic data.

Reply:
Yeah, we've never really figured out best practices for factories and seeders and it can be a painful at times.  It seems like we're always in a spot where way too much is being automatically created or it takes like 10 steps to get a small piece of data to properly seed.  :sweat: 

If you want 0-3 randomly instead of 3 that seems fine to me.